### ISSUE TITLE
Rails framework treats me like i'm stupid, but i'm not (i hope)

### ISSUE THREAD

**Issue by nleo** [rails/rails#51471]
`Post.order("length(title)").first`

vs

`Post.order(Arel.sql("length(title)")).first`

I choosed ruby and rails laconical, expressive. Write less - gets more. Second line is not rails way.

Docs says it workaround, like kludge. You put unready raw checks in stable and remove a way to disable it.

```
def self.allow_unsafe_raw_sql=(value) # :nodoc:
  ActiveSupport::Deprecation.warn("ActiveRecord::Base.allow_unsafe_raw_sql= is deprecated and will be removed in Rails 7.0")
end
```

I have single user, local deployed, pet project. I want to disable this check. I don't want to type useless overabundant symbols every time I need order things.

I'm not stupid, I know exactly what I'm doing.

**Comment by nleo** [comment]
You can't protect coders from all mistakes.  Price too high.

There are static code analyzers, warnings etc. Its not area of framework responsability

**Comment by fatkodima** [comment]
Since this is not a rails issue, better to open a discussion in the rails forum (https://discuss.rubyonrails.org) or in the discord channel.

I am not stupid too, but these checks were added for a reason. When many people are working together or you can forget by yourself and accidentally via a bunch of mixed up method calls, for example, pass a user provided input to this and get a potential sql injection.

**Comment by nleo** [comment]
@fatkodima thanks for reply

> this is not a rails issue

Which issue is it? Ruby?

> hese checks were added for a reason

Can you provide links to discussions? I can't believe whole rails team thinks it's good idea to mix static code analyzing features inside ORM and make it mandatory is a good idea.

> discussion in the rails forum (https://discuss.rubyonrails.org/) or in the discord

Sorry, I don't use this things and very comfortable with github issues for code and development talks

**Comment by nleo** [comment]
I just so much disapointing that rails start targeting housewifes users nor professional programmers, so in nearest future I have to start loking for replacement. So sad.

**Comment by rafaelfranca** [comment]
Have you heard about a method called `html_safe`? It was introduced in Rails 2.3, 14 years ago.

How about Strong Parameters? It was introduced in Rails 4, 11 years ago.

How about the entirety of Active Record that already deal with SQL injection for you most of the time?

This feature is on the same helm, it is part of the goal of the framework of making security easy. It has nothing to do with non-professional programmers. Everyone can make insecure applications and Rails always lead the charge of making security easy.

We aren't planning to change this feature in the near feature. Maybe later we will make it less verbose, and we are always open for suggestions on how to make this more friendly, but so far, we had no good suggestions.

In summary, accepting arbitrary strings in methods that can lead to SQL injection, one of the top 3 most common security problem in every application in the web, isn't something that Rails will do anymore.

**Comment by nleo** [comment]
@rafaelfranca thank you for so expanded answer

1. `where("a = '#{myvar}'")` - don't raise error

and as experienced developer I know why this call risky and why I should use `where("a = ?", user_input_var)`

2. Strong Parameters. I dont remeber can I disable it at all, but it defenetly have easy and not annoying way to bay pass.

3. Also I know what is `config.require_master_key=false` and disable on each new rails app. Couse I dont need this.

4. Besides it guarantees nothing. I can puts `Arel.sql` everywhere in big project now as it needed, but if I have a big team with messy code I still cat get sql injection after time, becouse somebody can add user input there it wasnt before and foget about `Arel.sql` bypass.

I know you want better for rails developers, but more I think about way you trying to make it, less I like it. Sorry for inconvenience.

**Comment by nleo** [comment]
Also want to tell story about security.

I saw a project which have low bugets and a bunch of low skill low motivated freelacers pass throw it. Some gems I think 10 years without updates. Nobody reads patchnotes, CSV news. Very messy, a lot of dependences. Everybody add his favorite gems. I think there is a lot of holes in app, but nobody cares, nobody intrested to purposefully hack it. So it lives and generates money income.

So you can't force people to write secure code, while they not motivated by themselfs. You can get only annoyance and "Ok, I'll put `Arel.sql` here, not understand why for what I'm doing this, but ok. I just want to finish my job and get money"

And you can do nothing with this with coding things.

**Comment by nleo** [comment]
@rafaelfranca I want explain why this feature trigger me so much.

Time ago I saw a react devs write and sustain code like this:

```javascript
HTTP_GET_SOMEENDPOINT1 = 'http_get_someendpoint1'
HTTP_PUT_SOMEENDPOINT1 = 'http_put_someendpoint1'
HTTP_GET_SOMEENDPOINT2 = 'http_get_someendpoint2'
```

I looks for me so raw so unready technology. Technology that force developers to do so monkey job. So I decide that I'm not intrested to work with react.

Putting `Arel.sql` between `order` and plain string literal like `width*height` looks for me same monkey job.

Rails reached very **high standarts** and reputation of **api quality** and developer compfort. And **this feature** much **much lower** then that.

Rails free and you do it for free. I'm appreciate. You don't have to meet my expectations. I understand.

But still it's below your stantarts of quality.
